src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Parsing/StringsHelper.cs

```diff
@@ -119,5 +119,34 @@ public static string RemoveBracketsQuotesParenthesis(string value)
 
             return value;
         }
+
+        public static string ToSnakeCase(this string text)
```

Comment:
You might want to combine this with [ConvertToUpperSnakeCase](https://github.com/newrelic/newrelic-dotnet-agent/blob/a77919e5405283fc57ed0251b731c5c1f3047305/src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Providers/Wrapper/Constants.cs#L125)

Reply:
I did look at that initially, but that code in that class seems very specific to converting Enum members to strings.  I'll make a note to look at it again before merging the feature branch to main.